ovs/dal/hybrids/storagedriver.py

```diff
@@ -187,6 +189,28 @@ def _cluster_node_config(self):
         """
         from ovs.extensions.generic.configuration import Configuration
         rdma = Configuration.get('/ovs/framework/rdma')
+        distance_map = {}
+        primary_domains = []
+        secondary_domains = []
+        for junction in self.storagerouter.domains:
+            if junction.backup is False:
+                primary_domains.append(junction.domain_guid)
+            else:
+                secondary_domains.append(junction.domain_guid)
+        for sd in self.vpool.storagedrivers:
+            if sd.guid == self.guid:
+                continue
+            if len(primary_domains) == 0:
+                distance_map[str(sd.storagedriver_id)] = StorageDriver.DISTANCES.NEAR
+            else:
+                distance = StorageDriver.DISTANCES.INFINITE
+                for junction in sd.storagerouter.domains:
+                    if junction.backup is False:
+                        if junction.domain_guid in secondary_domains:
+                            distance = min(distance, StorageDriver.DISTANCES.FAR)
+                        if junction.domain_guid in primary_domains:
+                            distance = min(distance, StorageDriver.DISTANCES.NEAR)
```

Comment:
Reversing these if statements and breaking when finding the NEARest, is more optimal